src/main/java/edu/ucla/library/iiif/auth/services/DatabaseServiceImpl.java

```diff
@@ -0,0 +1,209 @@
+package edu.ucla.library.iiif.auth.services;
+
+import java.net.URI;
+
+import edu.ucla.library.iiif.auth.Config;
+import edu.ucla.library.iiif.auth.MessageCodes;
+
+import info.freelibrary.util.Logger;
+import info.freelibrary.util.LoggerFactory;
+import info.freelibrary.util.StringUtils;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.pgclient.PgConnectOptions;
+import io.vertx.pgclient.PgPool;
+import io.vertx.sqlclient.PoolOptions;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowSet;
+import io.vertx.sqlclient.SqlClient;
+import io.vertx.sqlclient.Tuple;
+
+/**
+ * The implementation of DatabaseService.
+ */
+public class DatabaseServiceImpl implements DatabaseService {
+
+    /**
+     * The database service's logger.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatabaseService.class, MessageCodes.BUNDLE);
+
+    /**
+     * The postgres database (and default user) name.
+     */
+    private static final String POSTGRES = "postgres";
+
+    /**
+     * The name of the database's "items" table.
+     */
+    private static final String ITEMS = "items";
+
+    /**
+     * The name of the database's "origins" table.
+     */
+    private static final String ORIGINS = "origins";
+
+    /**
+     * A single space.
+     */
+    private static final String SPACE = " ";
```

Comment:
I believe this exists in freelib-utils' Constants class: `Constants.SPACE` (which is a dependency of hauth).